src/formatDate.js

```diff
@@ -50,7 +50,26 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat.pop();
+  const dateParts = date.split(separator);
+  const dateMap = {};
+
+  fromFormat.forEach((format, index) => {
+    dateMap[format] = dateParts[index];
+  });
+
+  if (dateMap['YYYY'] && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
```

Comment:
The condition `dateMap['YYYY'] && toFormat.includes('YY')` is not sufficient to ensure that the 'YY' format is not already present in the `dateMap`. It's important to check that 'YY' is not a key in `dateMap` before assigning a new value to avoid overwriting existing data.